kernel/pci/src/lib.rs

```diff
@@ -211,84 +249,133 @@ pub struct PciLocation {
     func: u8,
 }
 
+struct PciAddress {
+    /// Offset to a double word in a PCI configuration space
+    dword_address: u32,
+    /// Bit offset of the bytes of interest in this dword
+    byte_shift: u32,
```

Comment:
this documentation is needed, but it's still a bit confusing. It's a bitwise shift value, so what does it have to do with "byte"s?

Reply:
Then perhaps we should use an enum instead, especially if this field should be restricted to a certain subset of values. perhaps:
```rust
/// todo: Docs, plus think of a better name, lol
enum ByteSelector {
    Byte0, // = 0
    Byte1, // = 8,
    Byte2, // = 16,
    Byte3, // = 32,
}
```

then each variant can either be given a `u8` value (as shown by the suggested comments) or you can implement a function on that enum that returns a bitshift value or extracts the actual byte(s) from an inputted `u32`.

Reply:
bump